Libs/MRML/Core/vtkMRMLModelDisplayNode.h

```diff
@@ -73,23 +90,48 @@ class VTK_MRML_EXPORT vtkMRMLModelDisplayNode : public vtkMRMLDisplayNode
   /// \sa SetActiveScalarName()
   virtual void SetActiveAttributeLocation(int location);
 
-  /// Update the pipeline based on this node attributes
-  virtual void UpdatePolyDataPipeline();
+  /// Reimplemented to update scalar range accordingly
+  /// \sa SetActiveScalarName()
+  virtual void SetScalarRangeFlag(int flag);
+
+  /// Set whether to threshold the model display node.
+  /// \sa ThresholdEnabled, GetThresholdEnabled()
+  vtkSetMacro(ThresholdEnabled,bool);
+  vtkBooleanMacro(ThresholdEnabled,bool);
+
+  /// Get whether to threshold the model display node.
+  /// \sa ThresholdEnabled, SetThresholdEnabled(), ThresholdEnabledOn(),
+  /// ThresholdEnabledOff()
+  vtkGetMacro(ThresholdEnabled,bool);
+
+  /// Set the threshold range of the model display node.
+  /// \sa GetThresholdMin(), GetTresholdMax()
+  void SetThresholdRange(double min, double max);
+  void SetThresholdRange(double range[2]);
+
+  /// Get the threshold range of the model display node.
+  /// \sa SetThresholdRange()
+  void GetThresholdRange(double range[2]);
+  double GetThresholdMin();
+  double GetThresholdMax();
 
 protected:
   vtkMRMLModelDisplayNode();
   ~vtkMRMLModelDisplayNode();
   vtkMRMLModelDisplayNode(const vtkMRMLModelDisplayNode&);
   void operator=(const vtkMRMLModelDisplayNode&);
 
-  virtual void ProcessMRMLEvents(vtkObject *caller,
```

Comment:
When running model tests and debugging, it appeared that `UpdatePolyDataPipeline()` (now `UpdateAssignedAttribute()`) was called literally a hundred times before a mesh would even be set and observed by the model node, since that `ProcessMRMLEvents()` would just call it everytime an object of the scene calls `ModifiedEvent` (from my understanding?)
Instead, I only called it when `SetActiveScalarName` and `SetActiveScalarLocation` were changed, and would call `AssignAttribute::Update()` to ensure the changes would propagate through the pipeline. Also, if the model node is modified, that means the input mesh is modified, and changes should automatically be propagated along the pipeline.

Even though I first removed it for convenienve during debugging, the tests for the models kept working, as well as the Models module GUI and its event propagation, so I figured it would be a good performance increase not to run that pipeline which appeared to often run for no good reason... but I didn't thoroughly test all the remaining tests so if there are failing ones I must be missing someone in my reasoning :)